app/src/main/java/dev/ukanth/ufirewall/log/LogRecyclerViewAdapter.java

```diff
@@ -107,42 +98,37 @@ private Bitmap getAppIcon26(PackageManager mPackageManager, ApplicationInfo appl
 
     @Override
     public void onBindViewHolder(ViewHolder holder, int position) {
-        data = logData.get(position);
+        LogData data = logData.get(position);
         PackageManager manager = context.getPackageManager();
-        holder.bind(logData.get(position),recyclerItemClickListener);
+        holder.bind(logData.get(position), recyclerItemClickListener);
         try {
-            HashMap<Integer,String> listMaps = Api.getPackagesForUser(Api.getListOfUids());
-            info = Api.getPackageDetails(context, listMaps, data.getUid());
-            if(info!=null && info.applicationInfo !=null) {
-                Object drawable = info.applicationInfo.loadIcon(manager);
-                if (drawable instanceof Bitmap)
```

Comment:
Why this logic is removed ?

Reply:
The ' drawable instanceof Bitmap' is always false (note that loadIcon() always returns a Drawable).

Reply:
It was added due to a crash on a specific devices.

Reply:
Hummm, it probably fixed the crash by always falling into the else case.
I've noticed some inconsistencies around that area of the code related to different handling on different API levels.

Reply:
I'm going to revert that change on this branch and make a new pull for this.